src/persistent/cPersistence.c

```diff
@@ -781,7 +813,11 @@ Per_dealloc(cPersistentObject *self)
     }
 
     if (self->cache)
-        cPersistenceCAPI->percachedel(self->cache, self->oid);
+    {
+        capi_struct = _get_capi_struct(Py_TYPE((PyObject*)self));
+        if (capi_struct != NULL)
+            capi_struct->percachedel(self->cache, self->oid);
+    }
     Py_XDECREF(self->cache);
     Py_XDECREF(self->jar);
     Py_XDECREF(self->oid);
```

Comment:
We need to decref the type too, when using heap types https://docs.python.org/3.11/howto/isolating-extensions.html#defining-tp-dealloc

Reply:
64831e92